server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchDocValuesContext.java

```diff
@@ -33,20 +32,25 @@
 
     private final List<FieldAndFormat> fields;
 
-    public static FetchDocValuesContext create(Function<String, Set<String>> simpleMatchToFullName,
-                                               int maxAllowedDocvalueFields,
-                                               List<FieldAndFormat> fieldPatterns) {
+    public static FetchDocValuesContext create(MapperService mapperService, List<FieldAndFormat> fieldPatterns) {
+        int maxAllowedDocvalueFields = mapperService.getIndexSettings().getMaxDocvalueFields();
+
         List<FieldAndFormat> fields = new ArrayList<>();
+        int mappedFieldCount = 0;
         for (FieldAndFormat field : fieldPatterns) {
-            Collection<String> fieldNames = simpleMatchToFullName.apply(field.field);
+            Collection<String> fieldNames = mapperService.simpleMatchToFullName(field.field);
             for (String fieldName: fieldNames) {
                 fields.add(new FieldAndFormat(fieldName, field.format));
+                // only count the mapped fields towards the limit fedinde by IndexSettings.MAX_DOCVALUE_FIELDS_SEARCH_SETTING
+                if (mapperService.fieldType(fieldName) != null) {
```

Comment:
For efficiency could we only create a new `FieldAndFormat` if the field is mapped? I think we'll end up dropping the field anyways, not even having an entry in the response for it.

Also small typo, fedinde -> defined.

Reply:
You are right, I only see the "fields" really being used in "FetchDocValuePhase#getProcessor" and we seem to disregard fields without a mapping there. I will update the comments on the class though to reflect this for future usage.